### PR TITLE
Adopt icon font in outline/breadcrumbs/suggest widget

### LINKED ISSUE #78761: New arrow icon cannot be shown completely
image

For this, I zoomed the VS Code with <kbd>Ctrl</kbd>+<kbd>+</kbd> and <kbd>Ctrl</kbd>+<kbd>-</kbd>. In the following 4 situations, the second one and third one's arrow icons are broken. 

### zoom level 1
image

### zoom level 2
image

### zoom level 3
image

### zoom level 4
image

The old triangle works fine but now was replaced by an arrow, which is broken in some situations. Hope it can be fixed in later versions.

- VSCode Version: 1.37.0
- OS Version: deepin 15.11

Steps to Reproduce:

1. Upgrade VS Code to 1.37.0 version
2. Found that the arrow icon is broken

Does this issue occur when all extensions are disabled?: Yes

### PR DESCRIPTION
**Pull request by miguelsolorio**
Related #78889 and fixes #78761

This adopts the icon font in the outline/breadcrumbs/suggest widget:

image

image

image

This also introduces color tokens for all of the symbol icons:

- `symbolIcon.arrayForeground`
- `symbolIcon.booleanForeground`
- `symbolIcon.classForeground`
- `symbolIcon.contstantForeground`
- `symbolIcon.constructorForeground`
- `symbolIcon.enumeratorForeground`
- `symbolIcon.enumeratorMemberForeground`
- `symbolIcon.eventForeground`
- `symbolIcon.fieldForeground`
- `symbolIcon.fileForeground`
- `symbolIcon.functionForeground`
- `symbolIcon.interfaceForeground`
- `symbolIcon.keyForeground`
- `symbolIcon.methodForeground`
- `symbolIcon.moduleForeground`
- `symbolIcon.namespaceForeground`
- `symbolIcon.nullForeground`
- `symbolIcon.numberForeground`
- `symbolIcon.objectForeground`
- `symbolIcon.operatorForeground`
- `symbolIcon.packageForeground`
- `symbolIcon.propertyForeground`
- `symbolIcon.stringForeground`
- `symbolIcon.structForeground`
- `symbolIcon.typeParameterForeground`
- `symbolIcon.variableForeground`

And when configured, can look like this:

image

cc @aeschli for color token names

### CODE REVIEW

File: src/vs/editor/contrib/documentSymbols/outlineTree.ts [hunk 1]
```diff
@@ -273,3 +273,146 @@ export class OutlineDataSource implements IDataSource<OutlineModel, OutlineItem>
...
+	dark: '#75BEFF',
+	light: '#007ACC',
+	hc: '#75BEFF'
+}, localize('outlineInterfaceSymbol.foreground', 'The foreground color for interface symbols. These symbols appear in the outline, breadcrumb, and suggest widget.'));
+
+export const OUTLINE_VARIABLE_SYMBOL_FOREGROUND = registerColor('outlineVariableSymbol.foreground', {
+	dark: '#75BEFF',
+	light: '#007ACC',
+	hc: '#75BEFF'
+}, localize('outlineVariableSymbol.foreground', 'The foreground color for variable symbols. These symbols appear in the outline, breadcrumb, and suggest widget.'));
```
Comment (jrieken) [inline src/vs/editor/contrib/documentSymbols/outlineTree.ts]: This doesn't look like a complete list of the symbols we have?

**Comment by beastdestroyer** [comment]
What does that blank-ish square in set/settimeout mean? Personally I don't find it that meaningful

**Comment by usernamehw** [comment]
Snippet. https://code.visualstudio.com/docs/editor/intellisense#_types-of-completions

File: src/vs/editor/contrib/documentSymbols/outlineTree.ts (hunk 1 above)
Comment (misolori) [reply src/vs/editor/contrib/documentSymbols/outlineTree.ts]: I only included the ones that have color, the rest default to “outlineDefaultSymbol.foreground”. Are you thinking we’d define a token for each symbol type?

Comment (jrieken) [reply src/vs/editor/contrib/documentSymbols/outlineTree.ts]: You mean those that come with a color by default, right? I feel like folks would expect to be able to change the colour of all icons.

Comment (jrieken) [reply src/vs/editor/contrib/documentSymbols/outlineTree.ts]: Yeah, makes sense

**Review by jrieken — APPROVED** [review]

**Comment by misolori** [comment]
@jrieken updated the pr with a minor change. The icons needed to be wrapped into their own container/element as the parent container with the `.codicon` class was affecting all child font properties, including the label. If you could take another quick look that'd be great.

**Comment by aeschli** [comment]
@misolori I'd suggest the names
`symbolIcon.methodForeground`, `symbolIcon.classForeground` ...

I wouldn't define a color for `symbolIcon.defaultForeground`, but just use the default foreground color for it.

**Comment by misolori** [comment]
Forgot to update the main description, there is no default and each symbol kind has a color token. I also updated the naming convention based on your suggestions.

image

**Comment by misolori** [comment]
@jrieken ping 😄

**Review by jrieken — APPROVED** [review]
ops - 🚢 it

**Pull request merged**